src/index.js

```diff
@@ -70,23 +71,30 @@ const elmWebComponents = {
           const flags = mapFlags(props)
           context.flags = flags
 
+          var elmDiv = this;
+          var parentDiv = this;
+
+          if (useShadowDom) {
+            parentDiv = this.attachShadow({mode: 'open'});
+          }
+
           if (elmVersion === '0.19') {
             /* a change in Elm 0.19 means that ElmComponent.init now replaces the node you give it
              * whereas in 0.18 it rendered into it. To avoid Elm therefore destroying our custom element
              * we create a div that we let Elm render into, and manually clear any pre-rendered contents.
              */
-            const elmDiv = document.createElement('div')
+            elmDiv = document.createElement('div')
 
-            this.innerHTML = ''
-            this.appendChild(elmDiv)
+            parentDiv.innerHTML = ''
+            parentDiv.appendChild(elmDiv)
 
             const elmElement = ElmComponent.init({
               flags,
               node: elmDiv,
             })
             setupPorts(elmElement.ports)
           } else if (elmVersion === '0.18') {
-            const elmElement = ElmComponent.embed(this, flags)
+            const elmElement = ElmComponent.embed(elmDiv, flags)
```

Comment:
Should this not be `parentDiv`? Or if Elm rendering on 0.18 can't work, we should log a warning here if the flag was active

Reply:
That's probably a bug, good catch. I'll update the PR but I don't have time to test the change right now. I'll try it out during the week.

Reply:
I confirm that it was a bug and that the last change fixed it.